scripts/util/math.js

```diff
@@ -0,0 +1,141 @@
+/**
+ * Utility functions for Javascript.
+ * Could be ported to C++ and exposed globally in the future.
+ */
+
+/**
+ * 2D length of input vector
+ * @param {object} vec
+ * @returns {number}
+ */
+function getSize(vec) {
+	return Math.sqrt(getSizeSquared(vec));
+}
+
```

Comment:
This is worth doing in TS as well, I think it's so self-explanatory you should be able to do it even if you've not used TS before. You just need
```ts
type Vec2D = [number, number];
type Vec3D = [number, number, number];
```
though maybe worth porting those in `index.d.ts`, that'd need a separate PR into the submodule though so I guess just leave here for now. You can remove JSDoc type annoations if the args and return types are typed.

Reply:
Converted these. Very straightforward